<!DOCTYPE html>
<html lang="en">

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>【microRNA软件】miranda</title>
  <meta name="description" content="miranda - Finds potential target sites for miRNAs in genomic sequence">

  <link rel="stylesheet" href="/assets/main.css">
  <link rel="canonical" href="http://localhost:4040/biology/2017/03/18/miranda.html">
  <link rel="alternate" type="application/rss+xml" title="x2yline&#39;s blog" href="/feed.xml">
  
  
</head>


  <body>

    <header class="site-header" role="banner">

  <div class="wrapper">
    
    
    <a class="site-title" href="/">x2yline&#39;s blog</a>
  
    
      <nav class="site-nav">
        <input type="checkbox" id="nav-trigger" class="nav-trigger" />
        <label for="nav-trigger">
          <span class="menu-icon">
            <svg viewBox="0 0 18 15" width="18px" height="15px">
              <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
              <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
              <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
            </svg>
          </span>
        </label>

        <div class="trigger">
          
            
            
            <a class="page-link" href="/about/">About</a>
            
          
            
            
          
            
            
          
            
            
          
        </div>
      </nav>
    
  </div>
</header>


    <main class="page-content" aria-label="Content">
      <div class="wrapper">
        <article class="post" itemscope itemtype="http://schema.org/BlogPosting">

  <header class="post-header">
    <h1 class="post-title" itemprop="name headline">【microRNA软件】miranda</h1>
    <p class="post-meta">
      <time datetime="2017-03-18T09:24:05+08:00" itemprop="datePublished">
        
        Mar 18, 2017
      </time>
      </p>
  </header>

  <div class="post-content" itemprop="articleBody">
    <p><a href="http://www.microrna.org">miranda</a> - Finds potential target sites for miRNAs in genomic sequence</p>

<h4>简介</h4>
<p>miranda是历史上第一个研究出的miRNA靶基因预测软件，适用范围广，不受物种限制，允许G:U错配。在软件官网http://www.microrna.org 可以下载原始程序本地使用，也可以在线使用<font color="red">miRanda-mirSVR</font>系统（还有查看miRNA在不同组织中的<font color="red">表达谱数据</font>功能），下面分别介绍着两种使用方法。
&lt;/br&gt;</p>
<h4>使用方法1：下载程序本地使用</h4>
<p>下载C代码压缩包<a href="http://cbio.mskcc.org/microrna_data/miRanda-aug2010.tar.gz">click here</a>后，在linux系统下进行编译安装后（安装方法见README文件）
命令行输入：</p>
<figure class="highlight"><pre><code class="language-shell" data-lang="shell">miranda file1 file2 <span class="o">[</span>-sc score] <span class="o">[</span>-en energy] <span class="o">[</span>-scale scale] <span class="o">[</span>-strict] <span class="o">[</span>-go X] <span class="o">[</span>-ge Y] <span class="o">[</span>-out fileout] <span class="o">[</span>-quiet] <span class="o">[</span>-trim T] <span class="o">[</span>-noenergy] <span class="o">[</span>-restrict file]</code></pre></figure>

<p>中括号中是可选参数，file1 是RNA（miRNA）的 FASTA 格式文件，file2是某个基因的FASTA 格式的文件</p>

<p>该软件使用<font colot="blue">两步法</font>进行靶标的预测：
第一步是用动态规划算法进行两个序列之间的局部比对，基于互补结果进行打分，软件允许G:U的配对，不过分数比正常的配对低一点。
第二步是选择第一步中分数较高的比对结果（设置阈值，阈值以-sc表示），并对查询序列（miRNA），连接序列和倒转的参考序列的折叠结构热稳定性进行计算（计算出其最低自由能），计算方法来源于 Ivo Hofacker写的<font color="blue">ViennaRNA</font>包中的子程序RNAlib。
最后筛选出自由能小于阈值（记为-en）的靶序列，输出靶标序列位点和阈值信息。
一般选择比对分数大于<font color="blue">150</font>，自由能绝对值大于<font color="blue">7</font>即可（也有的选择比对分数大于155，自由能绝对值大于20）</p>
<font color="red">示例</font>
<p>如下：</p>

<figure class="highlight"><pre><code class="language-shell" data-lang="shell"><span class="o">[</span>root@localhost bin]# ./miranda /root/biosoft/miRNA/miRanda-3.3a/examples/bantam_stRNA.fasta /root/biosoft/miRNA/miRanda-3.3a/examples/hid_UTR.fasta

<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>
miranda v3.3a microRNA Target Scanning Algorithm
<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>
<span class="o">(</span>c<span class="o">)</span> 2003 Memorial Sloan-Kettering Cancer Center, New York

Authors: Anton Enright, Bino John, Chris Sander and Debora Marks
<span class="o">(</span>mirnatargets <span class="o">(</span>at<span class="o">)</span> cbio.mskcc.org - reaches all authors<span class="o">)</span>

Software written by: Anton Enright
Distributed <span class="k">for </span>anyone to use under the GNU Public License <span class="o">(</span>GPL<span class="o">)</span>,
See the files <span class="s1">'COPYING'</span> and <span class="s1">'LICENSE'</span> <span class="k">for </span>details

If you use this software please cite:
Enright AJ, John B, Gaul U, Tuschl T, Sander C and Marks DS;
<span class="o">(</span>2003<span class="o">)</span> Genome Biology; 5<span class="o">(</span>1<span class="o">)</span>:R1.

  miranda comes with ABSOLUTELY NO WARRANTY;
  This is free software, and you are welcome to redistribute it
  under certain conditions; <span class="nb">type</span> <span class="sb">`</span>miranda --license<span class="s1">' for details.

Current Settings:
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
Query Filename: /root/biosoft/miRNA/miRanda-3.3a/examples/bantam_stRNA.fasta
Reference Filename: /root/biosoft/miRNA/miRanda-3.3a/examples/hid_UTR.fasta
Gap Open Penalty: -9.000000
Gap Extend Penalty: -4.000000
Score Threshold: 140.000000
Energy Threshold: 1.000000 kcal/mol
Scaling Parameter: 4.000000
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
Read Sequence:gi|29565487|emb|AJ550546.1| Drosophila melanogaster microRNA miR-bantam(21 nt)
Read Sequence:gi|945100|gb|U31226.1|DMU31226 Drosophila melanogaster head involution defective protein (hid) mRNA, complete cds (3'</span>UTR only<span class="o">)(</span>3902 nt<span class="o">)</span>
<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>
Performing Scan: gi|29565487|emb|AJ550546.1| vs gi|945100|gb|U31226.1|DMU31226
<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>-<span class="o">=</span>

  Forward: Score: 167.000000 Q:2 to 20 R:3340 to 3360 Align Len <span class="o">(</span>18<span class="o">)</span> <span class="o">(</span>83.33%<span class="o">)</span> <span class="o">(</span>94.44%<span class="o">)</span>

  Query: 3<span class="s1">' gtCGAAAGTTTTACTAGAGTg 5'</span>
              |:||||| <span class="o">||||||||</span>|:
  Ref: 5<span class="s1">'   taGTTTTCACAATGATCTCGg 3'</span>

  Energy: -24.540001 kCal/Mol

Scores <span class="k">for </span>this hit:
&gt;gi|29565487|emb|AJ550546.1| gi|945100|gb|U31226.1|DMU31226 167.00 -24.54 2 20 3340 3360 18 83.33% 94.44%

  Forward: Score: 156.000000 Q:2 to 17 R:2505 to 2525 Align Len <span class="o">(</span>15<span class="o">)</span> <span class="o">(</span>86.67%<span class="o">)</span> <span class="o">(</span>93.33%<span class="o">)</span>

  Query: 3<span class="s1">' gtcgaAAGTTTTACTAGAGTg 5'</span>
                 <span class="o">||||||</span> <span class="o">||</span>:|||||
  Ref: 5<span class="s1">'   tcataTTCAAATTGGTCTCAc 3'</span>

  Energy: -20.030001 kCal/Mol

Scores <span class="k">for </span>this hit:
&gt;gi|29565487|emb|AJ550546.1| gi|945100|gb|U31226.1|DMU31226 156.00 -20.03 2 17 2505 2525 15 86.67% 93.33%

  Forward: Score: 155.000000 Q:2 to 16 R:2852 to 2872 Align Len <span class="o">(</span>14<span class="o">)</span> <span class="o">(</span>78.57%<span class="o">)</span> <span class="o">(</span>85.71%<span class="o">)</span>

  Query: 3<span class="s1">' gtcgaaAGTTTTACTAGAGTg 5'</span>
                  <span class="o">||</span>: | <span class="o">||||||||</span>
  Ref: 5<span class="s1">'   caaaaaTCGCATTGATCTCAt 3'</span>

  Energy: -14.570000 kCal/Mol

Scores <span class="k">for </span>this hit:
&gt;gi|29565487|emb|AJ550546.1| gi|945100|gb|U31226.1|DMU31226 155.00 -14.57 2 16 2852 2872 14 78.57% 85.71%

  Forward: Score: 152.000000 Q:2 to 18 R:3820 to 3841 Align Len <span class="o">(</span>17<span class="o">)</span> <span class="o">(</span>76.47%<span class="o">)</span> <span class="o">(</span>76.47%<span class="o">)</span>

  Query: 3<span class="s1">' gtcgAAAGT-TTTACTAGAGTg 5'</span>
                 | <span class="o">||</span>| | <span class="o">||||||||</span>
  Ref: 5<span class="s1">'   taaaTATCATTATTGATCTCAt 3'</span>

  Energy: -14.180000 kCal/Mol

Scores <span class="k">for </span>this hit:
&gt;gi|29565487|emb|AJ550546.1| gi|945100|gb|U31226.1|DMU31226 152.00 -14.18 2 18 3820 3841 17 76.47% 76.47%

Score <span class="k">for </span>this Scan:
Seq1,Seq2,Tot Score,Tot Energy,Max Score,Max Energy,Strand,Len1,Len2,Positions
&gt;&gt;gi|29565487|emb|AJ550546.1| gi|945100|gb|U31226.1|DMU31226 630.00 -73.32 167.00 -24.54 1 21 3902 3340 2505 2852 3820
Complete

Scan Complete</code></pre></figure>

<h4>使用方法2：在线网页工具</h4>

<p>在线使用的工具的比对方法和下载的软件相同，不过使用了<font color="blue">机器学习的算法</font>对miRNA和靶基因进行了评分，即<font color="red">miRanda-mirSVR</font>系统，综合高通量测序的数据进行评分，有两个主要评分结果</p>

<p>一是<font color="red">mirSVR score</font>表示对其下调的可能性进行评估，绝对值越大说明下调的越多，一般选择绝对值大于1
二是<font color="red">PhastCons score</font>表示对比对序列的保守性进行评分，分值越大越保守(详细介绍在 http://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-8-r90）</p>

<p>在线工具还有查看miRNA在人体不同组织的<font color="red">表达谱</font>功能（也可以直接下载所有表达谱数据），数据来源为cell上的一篇文章，数据来自172个人, 64个小鼠和16个大鼠的各个组织中的细胞：
Landgraf, Pablo, et al. “A mammalian microRNA expression atlas based on small RNA library sequencing.” Cell 129.7 (2007): 1401-1414.</p>

<p>参考：
https://www.biostars.org/p/5998/
http://www.microrna.org/microrna/faq.do</p>

  </div>

  
</article>

      </div>
    </main>

    <footer class="site-footer">

  <div class="wrapper">

    <h2 class="footer-heading">x2yline&#39;s blog</h2>

    <div class="footer-col-wrapper">
      <div class="footer-col footer-col-1">
        <ul class="contact-list">
          <li>
            
              x2yline&#39;s blog
            
            </li>
            
            <li><a href="mailto:1010006220@qq.com">1010006220@qq.com</a></li>
            
        </ul>
      </div>

      <div class="footer-col footer-col-2">
        <ul class="social-media-list">
          
          <li>
            <a href="https://github.com/x2yline"><span class="icon icon--github"><svg viewBox="0 0 16 16" width="16px" height="16px"><path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/></svg>
</span><span class="username">x2yline</span></a>

          </li>
          

          
          <li>
            <a href="https://twitter.com/x2yline"><span class="icon icon--twitter"><svg viewBox="0 0 16 16" width="16px" height="16px"><path fill="#828282" d="M15.969,3.058c-0.586,0.26-1.217,0.436-1.878,0.515c0.675-0.405,1.194-1.045,1.438-1.809c-0.632,0.375-1.332,0.647-2.076,0.793c-0.596-0.636-1.446-1.033-2.387-1.033c-1.806,0-3.27,1.464-3.27,3.27 c0,0.256,0.029,0.506,0.085,0.745C5.163,5.404,2.753,4.102,1.14,2.124C0.859,2.607,0.698,3.168,0.698,3.767 c0,1.134,0.577,2.135,1.455,2.722C1.616,6.472,1.112,6.325,0.671,6.08c0,0.014,0,0.027,0,0.041c0,1.584,1.127,2.906,2.623,3.206 C3.02,9.402,2.731,9.442,2.433,9.442c-0.211,0-0.416-0.021-0.615-0.059c0.416,1.299,1.624,2.245,3.055,2.271 c-1.119,0.877-2.529,1.4-4.061,1.4c-0.264,0-0.524-0.015-0.78-0.046c1.447,0.928,3.166,1.469,5.013,1.469 c6.015,0,9.304-4.983,9.304-9.304c0-0.142-0.003-0.283-0.009-0.423C14.976,4.29,15.531,3.714,15.969,3.058z"/></svg>
</span><span class="username">x2yline</span></a>

          </li>
          
        </ul>
      </div>

      <div class="footer-col footer-col-3">
        <p>个人博客, 笔记与软件收藏 :)
</p>
      </div>
    </div>

  </div>

</footer>


  </body>

</html>
